modules/workspace.nix

```diff
@@ -99,14 +108,27 @@ in
         Allows you to set wallpaper slideshow. Needs a directory of your wallpapers and an interval length.
       '';
     };
+
+    wallpaperPictureOfTheDay = lib.mkOption {
+      type = lib.types.nullOr wallpaperPictureOfTheDayType;
+      default = null;
+      example = "apod";
+      description = ''
+        Allows you to set wallpaper using the picture of the day plugin. Needs the provider and the fill mode.
+      '';
+    };
+
   };
 
   config = (lib.mkIf cfg.enable (lib.mkMerge [
     {
       assertions = [
         {
-          assertion = (cfg.workspace.wallpaperSlideShow == null || cfg.workspace.wallpaper == null);
-          message = "Cannot set both wallpaper and wallpaperSlideShow at the same time.";
+          assertion = ((cfg.workspace.wallpaperSlideShow == null && cfg.workspace.wallpaper == null && cfg.workspace.wallpaperPictureOfTheDay != null)
```

Comment:
Consider doing something like this instead: (see https://github.com/nix-community/home-manager/blob/master/modules/programs/git.nix#L360-L365)
```nix
assertion = let 
  wallpapers = with cfg.workspace; [wallPaperSlideShow wallpaper wallpaperPictureOfTheDay];
in
  lib.count (x: x != null) wallpapers <= 1;
```

Additionally, this assertion currently triggers when none of the three are set — are we really going to force people to pick a wallpaper?

Reply:
> ```nix
> assertion = let 
>   wallpapers = with cfg.workspace; [wallPaperSlideShow wallpaper wallpaperPictureOfTheDay];
> in
>   lib.count (x: x != null) wallpapers <= 1;
> ```

This! I really hated the way I did it, but I just couldn't figure out how to make it nicer.

> Additionally, this assertion currently triggers when none of the three are set — are we really going to force people to pick a wallpaper?

Oops. that definitely wasn't what I wanted to do.

Big thanks for the help